### PR TITLE
Rebase ghaf release 24.06

### PR DESCRIPTION
**Pull request by buianhhuy96**
- Update flake.lock to ghaf 24.06
- Modify config-processor to use ghaf modules
- Rename microvm interface id from "vm-*" to "tap-*"
- Simplify some configs due to updated nixpkgs
- Adapt app changes to new version: lisgd, squeekboard
- Remove upstream package overlay which caused error in previous build
- Remove local golang, go-app is now built with nixpkgs.buildGoModules
- Issue remained: fmo-dci warning and added unused (not working) audiovm

### CODE REVIEW

File: config-processor-installers.nix [hunk 1]
```diff
@@ -97,7 +98,7 @@ let
             isoImage.squashfsCompression = "lz4"; 
           }
         ]
-        ++ (import "${ghafOS}/modules/module-list.nix")
+        #++ (import "${ghafOS}/modules/module-list.nix")
```
Comment (jsvapiav) [inline config-processor-installers.nix:101]: Leftovers?

File: flake.nix [hunk 2]
```diff
@@ -64,7 +55,7 @@
 
       imports = [
         ./hydrajobs/flake-module.nix
-        ./modules/flake-module.nix
+        ./modules/flake-module.nix
```
Comment (jsvapiav) [inline flake.nix:58]: unnecessary space at the end of string

File: modules/fmo-services/dci-service/default.nix [hunk 3]
```diff
@@ -114,6 +114,8 @@ in {
       after = [
         "docker.service"
         "docker.socket"
+        # WAR: warning: fmo-dci.service is ordered after 'network-online.target' 
+        # but doesn't depend on it
```
Comment (jsvapiav) [inline modules/fmo-services/dci-service/default.nix:118]: Need to create a Jira ticket for that and put its number here.

File: config-processor-hardware.nix [hunk 4]
```diff
@@ -43,16 +43,23 @@ let
...
           self.nixosModules.fmo-configs
+          self.nixosModules.ghaf-common
+          ghafOS.nixosModules.host
 
-          (import "${ghafOS}/modules/host")
-          (import "${ghafOS}/modules/virtualization/microvm/microvm-host.nix")
+          (import "${ghafOS}/modules/microvm/networking.nix")
+          (import "${ghafOS}/modules/microvm/virtualization/microvm/microvm-host.nix")
+          
+          # WAR: ghaf mainline has audiovm hardcoded. This causes audiovm defined here
```
Comment (jsvapiav) [inline config-processor-hardware.nix:52]: Need to create a Jira ticket for that. And put its number here.

File: config-processor-installers.nix (hunk 1 above)
Comment (buianhhuy96) [reply config-processor-installers.nix]: Fixed.

File: flake.nix (hunk 2 above)
Comment (buianhhuy96) [reply flake.nix]: Fixed.

File: modules/fmo-services/dci-service/default.nix (hunk 3 above)
Comment (buianhhuy96) [reply modules/fmo-services/dci-service/default.nix:119]: Created FMO-44 and commented

File: config-processor-hardware.nix (hunk 4 above)
Comment (buianhhuy96) [reply config-processor-hardware.nix]: Created FMO-43 and commented

File: modules/packages/registration-agent/registration-agent-laptop.nix [hunk 5]
```diff
@@ -21,12 +11,9 @@ buildGo121Module {
   };
   tags = [ "prod" ];
   patches = [./remove-test.patch];
-  vendorSha256 = "sha256-9/twQyt6SVXWTRypt1FIWsRQxQEWFBkdi8eR+/xYNqg==";
+  vendorHash = "sha256-18p7l1otlviZNlM0UlCgW/US5YckBYcY/OEJoJIsIM0=";
```
[the remark this answers is not in the document]
Comment (buianhhuy96) [reply modules/packages/registration-agent/registration-agent-laptop.nix:14]: This is the updated hash code due to golang (`buildGoModule` in nixpkgs) updated.
RA version is not changed.

[the remark this answers is not in the document]
Comment (jsvapiav) [reply modules/packages/registration-agent/registration-agent-laptop.nix:14]: Cool, thanks!

**Review by jsvapiav — APPROVED** [review]
LGFM. Ship it!

**Pull request merged**